src/lib/core/WeaveExchangeMgr.cpp

```diff
@@ -1728,28 +1723,45 @@ void WeaveExchangeManager::WRMPStartTimer()
 
     if (foundWake) {
         // Set timer for next tick boundary - subtract the elapsed time from the current tick
-        int32_t timerArmValue = nextWakeTime * mWRMPTimerInterval - (System::Timer::GetCurrentEpoch() - mWRMPTimeStampBase);
+        System::Timer::Epoch currentTime = System::Timer::GetCurrentEpoch();
+        int32_t timerArmValue = nextWakeTime * mWRMPTimerInterval - (currentTime - mWRMPTimeStampBase);
+        System::Timer::Epoch timerExpiryEpoch = currentTime + timerArmValue;
+        System::Timer::Epoch prevTimerExpiryEpoch = 0;
```

Comment:
Based on what I'm reading here, `prevTimerExpiryEpoch` should perhaps be moved to be a member variable?